src/main/java/com/redhat/devtools/lsp4ij/features/documentSymbol/LSPBreadcrumbsRefreshListener.java

```diff
@@ -0,0 +1,111 @@
+import com.intellij.ui.components.breadcrumbs.Crumb;
+import com.intellij.xml.breadcrumbs.BreadcrumbListener;
+import com.intellij.xml.breadcrumbs.BreadcrumbsPanel;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.LanguageServerWrapper;
+import com.redhat.devtools.lsp4ij.ServerStatus;
+import com.redhat.devtools.lsp4ij.lifecycle.LanguageServerLifecycleListener;
+import com.redhat.devtools.lsp4ij.lifecycle.LanguageServerLifecycleManager;
+import kotlin.Unit;
+import kotlin.coroutines.Continuation;
+import org.eclipse.lsp4j.jsonrpc.MessageConsumer;
+import org.eclipse.lsp4j.jsonrpc.messages.Message;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Refreshes the breadcrumbs and sticky lines for open editor tabs when a language server has started.
+ */
+public class LSPBreadcrumbsRefreshListener implements ProjectActivity, LanguageServerLifecycleListener {
+
+    private static final Key<Boolean> ADDED_BREADCRUMB_LISTENER = Key.create(LSPBreadcrumbsRefreshListener.class + ".ADDED_BREADCRUMB_LISTENER");
+    private static final Key<Boolean> NEEDS_RESTART = Key.create(LSPBreadcrumbsRefreshListener.class + ".NEEDS_RESTART");
+
+    @Override
+    @Nullable
+    public Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
+        LanguageServerLifecycleManager.getInstance(project).addLanguageServerLifecycleListener(this);
+        return null;
+    }
+
+    @Override
+    public void handleStatusChanged(LanguageServerWrapper languageServer) {
+        if (languageServer.getServerStatus() == ServerStatus.started) {
+            Project project = languageServer.getProject();
+            for (VirtualFile virtualFile : FileEditorManager.getInstance(project).getOpenFiles()) {
+                PsiFile file = LSPIJUtils.getPsiFile(virtualFile, project);
+                if ((file != null) &&
+                    languageServer.getClientFeatures().getBreadcrumbsFeature().isSupported(file) &&
+                    languageServer.getClientFeatures().getBreadcrumbsFeature().isEnabled(file)) {
+                    for (Editor editor : LSPIJUtils.editorsForFile(virtualFile, project)) {
+                        BreadcrumbsPanel breadcrumbsComponent = BreadcrumbsPanel.getBreadcrumbsComponent(editor);
+                        if (breadcrumbsComponent != null) {
+                            // Only add a listener to the editor once
+                            if (editor.getUserData(ADDED_BREADCRUMB_LISTENER) == null) {
+                                breadcrumbsComponent.addBreadcrumbListener(new LSPBreadcrumbListener(file, editor), breadcrumbsComponent);
+                                editor.putUserData(ADDED_BREADCRUMB_LISTENER, true);
+                            }
+
+                            // Denote the file as needing restart and queue an update to its breadcrumbs
+                            file.putUserData(NEEDS_RESTART, true);
```

Comment:
And this tells the event listener that the next `#breadcrumbsChanged` event for the file should result in restarting analysis on the file.